README.md

```diff
@@ -1,69 +1,273 @@
+
+You will need to take note of the Object ID of the App Registration that will be managed by the Operator and the Client/Application ID, Client Secret and Tenant ID of the App Registration that will be used to Authenticate.
+
+Once you have created the app registration you will need to give it the correct permissions. This can be done by clicking on the API permissions tab in the Azure portal whilst viewing the app registration. You then need to click on Add a permission and then Microsoft Graph, click on Application permissions and search for Application. Add the Application.ReadWrite.All permission and click Add permissions, then click the Grant admin consent button.
+
+#### Configuring the ReplyURLSync config
+To configure the sync config so the Operator knows how to Authenticate with Azure, which App Registration to update and what Ingresses and URLs it should be managing, you will need to configure a `ReplyURLSync` custom resource. Currently, there are 6 fields available to configure the sync:
+
+#### Deploying the Operator to a cluster
+
+The commands below will deploy the Custom Resource Definitions (CRDs), RBAC, the Operator, the replyURLSync and the example Ingress. The Operator should be fully operational after these commands have been executed.
+
+**Note:** The Makefile has the ability to build and push the container image manually, but there are GitHub Workflows in the `.github/workflows` folder that automate the process.
+
+1. Build and push the controller images to a container registry (this step can be skipped if you already have an image built and pushed):
+   
+   Replace `<some-registry>` with the container registry you would like to push the image to and `<tag>` with the tag to identify the image.
+
+   ```sh
+   make docker-build docker-push IMG=<some-registry>/reply-urls-operator:<tag>
+   ```
+
+2. Update the container image for the manager deployment:
+
+   Before deploying the Reply URLs Operator you will need to update the image being declared in the deployment file `config/manager/manager.yaml`. Update the `manager` container in the containers section of the file with the image you have built. 
+   
+   That section should look similar to the snippet below.
+
+   ```yaml
+         containers:
+         - name: manager
+           command:
+           - /manager
+           image: sdshmctspublic.azurecr.io/reply-urls-operator:prod-c4620b7-20220905093200
+           imagePullPolicy: Always
+   ```
+
+3. Create reply-urls-operator secret
+   Currently the App Registration's Client Secret for the Operator is set as an environment variable. You will need to create a secret called `reply-urls-operator` with a data object called `azure-client-secret`.
+
+   Command to create the secret manually:
+   ```sh
+    kubectl create secret -n admin generic reply-urls-operator --from-literal azure-client-secret=<client_secret>
```

Comment:
better to separate this into two lines with an env variable, it's easier to copy paste full snippets rather than having to edit it

```suggestion
    AZURE_CLIENT_SECRET=
    kubectl create secret -n admin generic reply-urls-operator --from-literal azure-client-secret="${AZURE_CLIENT_SECRET}"
```